### PR TITLE
Update readme generator and generate README.rst only when fragments changed

### PR DESCRIPTION
**Pull request by sbidoul**

### CODE REVIEW

File: src/oca_github_bot/tasks/main_branch_bot.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def _gen_addons_readme(org, repo, branch, cwd):
     _logger.info("oca-gen-addon-readme in %s/%s@%s", org, repo, branch)
     gen_addon_readme_cmd = [
         "oca-gen-addon-readme",
+        "--if-fragments-changed",
```
Comment (legalsylvain) [inline src/oca_github_bot/tasks/main_branch_bot.py:36]: If i understand correctly, the readme will not be updated, if a PR change only the manifest info (like module name, maintainers, licences, ...)
If yes, It is not desired, dont you think ?

Comment (sbidoul) [reply src/oca_github_bot/tasks/main_branch_bot.py:36]: The option name might be misleading but the manifest is included in the digest computation as well as the whole readme fragments directory:

https://github.com/OCA/maintainer-tools/blob/9a5d4cbe17a7fe9921e4157d74e64461019e03f3/tools/gen_addon_readme.py#L516-L520

Comment (legalsylvain) [reply src/oca_github_bot/tasks/main_branch_bot.py:36]: Oh. Should be --if-source-changed ?

Comment (sbidoul) [reply src/oca_github_bot/tasks/main_branch_bot.py]: Alrighty. Done :)

**Pull request merged**